public/video-ui/src/components/ScheduledLaunch/ScheduledLaunch.js

```diff
@@ -11,89 +17,338 @@ export default class ScheduledLaunch extends React.Component {
+    const hasPreventedPublication = embargo && embargo > 16693689600000;
+    return (
+      <ul className="scheduleOptions">
+        {!hasPreventedPublication && (
+          <li>
+            <button
+              className="btn btn--list"
+              onClick={() => this.onSelectOption('schedule')}
+              disabled={!video || videoEditOpen}
+            >
+              {scheduledLaunch ? 'Edit scheduled date' : 'Schedule'}
+            </button>
+          </li>
+        )}
+        {!hasPreventedPublication && (
+          <li>
+            <button
+              className="btn btn--list"
+              onClick={() => this.onSelectOption('embargo')}
+              disabled={!video || videoEditOpen}
+            >
+              {embargo ? 'Edit embargo' : 'Embargo until...'}
+            </button>
+          </li>
+        )}
+        {!embargo &&
+          !scheduledLaunch && (
+            <li>
+              <button
+                className="btn btn--list"
+                onClick={() => this.preventPublication()}
+                disabled={!video || videoEditOpen}
+              >
+                Prevent publication
+              </button>
+            </li>
+          )}
+        {hasPreventedPublication && (
+          <li>
+            <button
+              className="btn btn--list"
+              onClick={() => {
+                this.removeDate('embargo');
+                this.setState({ showScheduleOptions: false });
+              }}
+              disabled={!video || videoEditOpen}
+            >
+              Remove indefinite embargo
+            </button>
+          </li>
+        )}
+      </ul>
+    );
+  };
+
+  renderAlert = invalidDateError =>
+    invalidDateError && (
+      <span className="topbar__alert">{invalidDateError}</span>
+    );
 
   render() {
-    const { video, video: { contentChangeDetails }, videoEditOpen } = this.props;
-    const { selectedDate } = this.state;
+    const {
+      video,
+      video: { contentChangeDetails },
+      videoEditOpen,
+      hasPublishedVideoUsages
+    } = this.props;
+    const {
+      selectedScheduleDate,
+      selectedEmbargoDate,
+      showScheduleOptions,
+      actionType
+    } = this.state;
     const showDatePicker = this.state.showDatePicker && !videoEditOpen;
-    const isFutureDate = selectedDate && moment(selectedDate).isAfter(moment());
-    const scheduledLaunch = contentChangeDetails && contentChangeDetails.scheduledLaunch && contentChangeDetails.scheduledLaunch.date;
+    const invalidDateError = this.state.invalidDateError;
+    const scheduledLaunch =
+      contentChangeDetails &&
+      contentChangeDetails.scheduledLaunch &&
+      contentChangeDetails.scheduledLaunch.date;
+    const embargo =
+      contentChangeDetails &&
+      contentChangeDetails.embargo &&
+      contentChangeDetails.embargo.date;
+    const hasPreventedPublication = embargo && embargo > 16693689600000;
+
     return (
```

Comment:
I'd split what comes next into separate render functions e.g. renderPreventedPublication, renderDates, renderDataPicker. You could nest these functions as well and end up with less complicated statements as well when deciding which components to render.